File: app/src/main/java/ru/mamykin/foreignbooksreader/common/FileToAndroidFileMapper.java
package ru.mamykin.foreignbooksreader.common;

import java.io.File;
import java.util.ArrayList;
import java.util.List;

import ru.mamykin.foreignbooksreader.models.AndroidFile;

/**
 * Creation date: 5/29/2017
 * Creation time: 12:38 PM
 * @author Andrey Mamykin(mamykin_av)
 */
public class FileToAndroidFileMapper {
    public List<AndroidFile> map(List<File> files) {
        List<AndroidFile> androidFiles = new ArrayList<>(files.size());
        for (File file : files) {
            androidFiles.add(new AndroidFile(file));
        }
        return androidFiles;
    }
}

File: app/src/main/java/ru/mamykin/foreignbooksreader/common/FileToAndroidFileMapper.kt
package ru.mamykin.foreignbooksreader.common

import java.io.File
import java.util.ArrayList

import ru.mamykin.foreignbooksreader.models.AndroidFile

class FileToAndroidFileMapper {
    fun map(files: List<File>): List<AndroidFile> {
        val androidFiles = ArrayList<AndroidFile>(files.size)
        for (file in files) {
            androidFiles.add(AndroidFile(file))
        }
        return androidFiles
    }
}